### PR TITLE
Logging done late enough might happen when capture already stopped.

### PR DESCRIPTION
**Pull request by s0undt3ch**
```
22:56:26,997 [pytestsalt.utils:206 ][INFO    ][1024] Stopping process psutil.Process(pid=1228, name='/home/vampas/.d', started='22:55:17') and respective children: [psutil.Process(pid=1245, name='/home/vampas/.d', started='22:55:18'), psut
il.Process(pid=1262, name='/home/vampas/.d', started='22:55:19'), psutil.Process(pid=1263, name='/home/vampas/.d', started='22:55:19')]                                                                                                        
Error in atexit._run_exitfuncs:                                                                                                                                                                                                                
Traceback (most recent call last):                                                                                                                                                                                                             
  File "/home/vampas/.dotfiles/.ext/pyenv/versions/3.5.4/lib/python3.5/logging/__init__.py", line 861, in handle                                                                                                                               
    self.emit(record)                                                                                                                                                                                                                          
  File "/home/vampas/.dotfiles/.ext/pyenv/versions/3.5.4/envs/Salt-3.5.4-Develop/lib/python3.5/site-packages/_pytest/logging.py", line 625, in emit                                                                                            
    logging.StreamHandler.emit(self, record)
  File "/home/vampas/.dotfiles/.ext/pyenv/versions/3.5.4/lib/python3.5/contextlib.py", line 66, in __exit__
    next(self.gen)
  File "/home/vampas/.dotfiles/.ext/pyenv/versions/3.5.4/envs/Salt-3.5.4-Develop/lib/python3.5/site-packages/_pytest/capture.py", line 167, in global_and_fixture_disabled
    self.resume_global_capture()
  File "/home/vampas/.dotfiles/.ext/pyenv/versions/3.5.4/envs/Salt-3.5.4-Develop/lib/python3.5/site-packages/_pytest/capture.py", line 120, in resume_global_capture
    self._global_capturing.resume_capturing()
AttributeError: 'NoneType' object has no attribute 'resume_capturing'
```
- [x] Create a new changelog file in the `changelog` folder, with a name like `<ISSUE NUMBER>.<TYPE>.rst`. See [changelog/README.rst](https://github.com/pytest-dev/pytest/blob/master/changelog/README.rst) for details.
- [x] Target the `master` branch for bug fixes, documentation updates and trivial changes.

### CODE REVIEW

File: src/_pytest/capture.py [hunk 1]
```diff
@@ -117,7 +117,9 @@ def stop_global_capturing(self):
             self._global_capturing = None
 
     def resume_global_capture(self):
-        self._global_capturing.resume_capturing()
+        cap = getattr(self, "_global_capturing", None)
```
Comment (asottile) [inline src/_pytest/capture.py]: the `getattr` here should be unnecessary -- it's only `None` while python tears down the process (and not a missing attribute)

I'd also leave a comment here along the lines of:

```python
    # during teardown of the python process, attributes are `None`d
    if self._global_capturing is not None:
        self._global_capturing.resume_capturing()
```

**Comment by s0undt3ch** [comment]
@nicoddemus believe me, I tried, for sure not long enough or there would be a test case. Anyway, I wasn't able to reproduce this in a test case :\

**Comment by nicoddemus** [comment]
@RonnyPfannschmidt @asottile @blueyed any objections to get this in without an acceptance test?

File: changelog/4487.bugfix.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+During teardown of the python process, capture attributes are None'd. On rare occasions global capture could be attempted to resume while is't already reset.
```
Comment (blueyed) [inline changelog/4487.bugfix.rst]: s/is't/it's/

**Comment by nicoddemus** [comment]
> Not sure how difficult a test would be, but it certainly would be great to have for corner cases like this.

Definitely, would you like to give this a try? I've tried for a few minutes to get this error to blow up with `atexit`, but no dice.

File: changelog/4487.bugfix.rst (hunk 2 above)
Comment (asottile) [reply changelog/4487.bugfix.rst]: and probably ` ``None`` ` as well

Comment (asottile) [reply changelog/4487.bugfix.rst]: ` ``None``'d` I think is correct

**Comment by s0undt3ch** [comment]
Comment and changelog entry reworded.

**Comment by blueyed** [comment]
Thanks!

**Comment by s0undt3ch** [comment]
Anytime, and Thanks!

**Comment by RonnyPfannschmidt** [comment]
@s0undt3ch i believe someone found a minimal test in #4500

**Comment by s0undt3ch** [comment]
Awesome!

**Comment by s0undt3ch** [comment]
https://github.com/pytest-dev/pytest/pull/4501
